kube/src/tls.rs

```diff
@@ -0,0 +1,78 @@
+use crate::config::Der;
+use crate::Result;
+
+#[cfg(feature = "native-tls")]
+mod native_impl;
+mod rustls_impl;
+/// Abstraction layer for TLS implementations
+#[non_exhaustive]
+#[derive(Debug, Clone, Copy)]
+pub enum Tls {
+    /// Use `rustls`
+    #[cfg(feature = "native-tls")]
+    Native,
+    /// Use `openssl`
+    #[cfg(feature = "rustls-tls")]
+    Rustls,
+}
+
+impl Tls {
+    /// Creates a TLS backend.
+    /// This function only works when exactly one backend
+    /// was configured, otherwise it will panic. This function
+    /// is only intended to use in tests.
+    #[allow(unreachable_code)]
+    pub fn pick() -> Tls {
+        let mut backend_count = 0;
+        if cfg!(feature = "native-tls") {
+            backend_count += 1;
+        }
+        if cfg!(feature = "rustls-tls") {
+            backend_count += 1;
+        }
+        assert_eq!(backend_count, 1);
+        #[cfg(feature = "native-tls")]
+        return Tls::Native;
+        #[cfg(feature = "rustls-tls")]
+        return Tls::Rustls;
+        unreachable!()
```

Comment:
Because this is an interface for `Client` construction, this effectively pushes a compile time error to a runtime error?